inc/class-tmsc.php

```diff
@@ -1,106 +1,314 @@
+										) ),
+									) ),
+								) ),
+								'CN' => new \Fieldmanager_Textfield( __( 'Guide Term CN', 'tmsc' ) ),
+							),
+						) ),
+					),
+				) ),
+			),
+		) );
+		$fm->activate_submenu_page();
+	}
+
+	/**
+	 * Keep track of our custom DB system processors.
+	 * @param string. $name. Namespaced class name.
+	 * @return void.
+	 */
+	public function register_processor( $name = '' ) {
+		if ( empty( $name ) || 'all' === $name ) {
+			foreach ( tmsc_get_system_processors() as $processor_slug => $processor_class_slug ) {
+				if ( empty( self::$instance->processors[ $processor_class_slug ] ) ) {
+					self::$instance->processors[ $processor_class_slug ] = '\\TMSC\\Database\\Processors\\' . TMSC_SYSTEM_BUILD_CLASS_PREFIX . '\\' . TMSC_SYSTEM_BUILD_CLASS_PREFIX . '_' . $processor_class_slug . '_Processor';
+				}
+			}
+		} else {
+			if ( empty( self::$instance->processors[ $name ] ) ) {
+				self::$instance->processors[ $name ] = '\\TMSC\\Database\\Processors\\' . TMSC_SYSTEM_BUILD_CLASS_PREFIX . '\\' . TMSC_SYSTEM_BUILD_CLASS_PREFIX . '_' . $name . '_Processor';
 			}
+		}
+		return self::$instance->processors;
+	}
 
-			// The Zoninator internal post_types array has already been written and cached by this point.
-			// We must override it to enabled these post types for the recent posts dropdown and AJAX searches.
-			z_get_zoninator()->post_types = array_merge( z_get_zoninator()->post_types, $post_types );
+	/**
+	 * Migrate all objects objects.
+	 */
+	public function migrate( $args = array(), $assc_args = array() ) {
+		if ( ! defined( 'WP_IMPORTING' ) ) {
+			define( 'WP_IMPORTING', true );
+		}
+		foreach ( self::$instance->processors( $args ) as $processor ) {
+			if ( ! in_array( $processor->processor_type, self::$instance->get_child_processors(), true ) ) {
+				$processor->run();
+				tmsc_stop_the_insanity();
+			}
 		}
+	}
 
-		/**
-		 * Only display tmsc objects in our TMSConnect Zones.
-		 */
-		public function filter_zone_post_types( $args ) {
-			$active_zone_id = z_get_zoninator()->_get_request_var( 'zone_id', 0, 'absint' );
-			$tmsc_zones = wp_list_pluck( $this->default_zones, 'slug' );
-			if ( ! empty( $active_zone_id ) ) {
-				$zone = z_get_zone( $active_zone_id );
-				if ( in_array( $zone->slug, $tmsc_zones ) ) {
-					$args['post_type'] = array( 'tms_object' );
+	/**
+	 * Apply a function to all post_meta matching the given key and update the
+	 * meta value with the return value of the function.
+	 */
+	public function filter_post_meta( $args = array() ) {
+		global $wpdb;
+		$meta_key = array_shift( $args );
+		$function = array_shift( $args );
+		if ( is_callable( $function ) ) {
+			$ids = $wpdb->get_results(
+				$wpdb->prepare(
+					"SELECT post_id, meta_value from {$wpdb->postmeta} WHERE meta_key = %s",
+					$meta_key
+				)
+			);
+			foreach ( $ids as $meta ) {
+				$new_value = call_user_func_array( $function, array( $meta->post_id, $meta_key, $meta->meta_value ) );
```

Comment:
This seems pretty dangerous to call any user function here.  Maybe look into using a filter instead?